frontend/src/citizen-frontend/login/LoginPage.tsx

```diff
@@ -165,6 +159,87 @@ export default React.memo(function LoginPage() {
   )
 })
 
+const weakLoginForm = validated(
+  object({
+    username: string(),
+    password: string()
+  }),
+  (form) => {
+    if (form.username.length === 0 || form.password.length === 0) {
+      return 'required'
+    }
+    return undefined
+  }
+)
+
+const authWeakLoginResult = wrapResult(authWeakLogin)
+
+const WeakLoginForm = React.memo(function WeakLogin({
+  unvalidatedNextPath
+}: {
+  unvalidatedNextPath: string | null
+}) {
+  const i18n = useTranslation()
+  const [rateLimitError, setRateLimitError] = useState(false)
+
+  const nextUrl = useMemo(
+    () =>
+      unvalidatedNextPath
+        ? parseUrlWithOrigin(window.location, unvalidatedNextPath)
+        : undefined,
+    [unvalidatedNextPath]
+  )
+
+  const form = useForm(
+    weakLoginForm,
+    () => ({ username: '', password: '' }),
+    i18n.validationErrors
+  )
+  const { username, password } = useFormFields(form)
+  return (
+    <>
+      <Gap size="m" />
+      <form action="" onSubmit={(e) => e.preventDefault()}>
+        <FixedSpaceColumn spacing="xs">
+          {rateLimitError && (
+            <AlertBox message={i18n.loginPage.login.rateLimitError} />
+          )}
+          <InputFieldF
+            autoComplete="email"
+            bind={username}
+            placeholder={i18n.loginPage.login.email}
+            width="L"
+            hideErrorsBeforeTouched={true}
+          />
+          <InputFieldF
+            autoComplete="current-password"
+            bind={password}
+            type="password"
+            placeholder={i18n.loginPage.login.password}
+            width="L"
+            hideErrorsBeforeTouched={true}
+          />
+          <AsyncButton
+            primary
+            type="submit"
+            text={i18n.loginPage.login.link}
+            disabled={!form.isValid()}
+            onClick={() =>
+              authWeakLoginResult(form.state.username, form.state.password)
+            }
+            onSuccess={() => window.location.replace(nextUrl ?? '/')}
+            onFailure={(error) => {
+              if (error.statusCode === 429) {
```

Comment:
`rateLimitError` ei koskaan aseteta takaisin falseksi. En tiedä tarviiko sitä nyt tässä vaiheessa korjata vielä, kun tämä on tällänen "engineering preview"-versio.
```suggestion
              setRateLimitError(error.statusCode === 429)
```